File: src/cb372/intellij/idris/IdrisFileType.java
package cb372.intellij.idris;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class IdrisFileType extends LanguageFileType {

  public static final IdrisFileType INSTANCE = new IdrisFileType();

  private IdrisFileType() {
    super(IdrisLanguage.INSTANCE);
  }

  @NotNull
  @Override
  public String getName() {
    return "Idris file";
  }

  @NotNull
  @Override
  public String getDescription() {
    return "Idris file";
  }

  @NotNull
  @Override
  public String getDefaultExtension() {
    return "idr";
  }

  @Nullable
  @Override
  public Icon getIcon() {
    return IdrisIcons.FILE;
  }

}

File: src/cb372/intellij/idris/IdrisFileType.scala
package cb372.intellij.idris

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing._

object IdrisFileType {
  val Instance = new IdrisFileType
}

class IdrisFileType private() extends LanguageFileType(IdrisLanguage.Instance) {
  def getName: String = "Idris file"
  def getDescription: String = "Idris file"
  def getDefaultExtension: String = "idr"
  def getIcon: Icon = IdrisIcons.File
}